app/models/agents/http_status_agent.rb

```diff
@@ -46,29 +50,47 @@ def validate_options
       errors.add(:base, "a url must be specified") unless options['url'].present?
     end
 
+    def header_array(str)
+      str.split(',').map { |el| el.strip }
+    end
+
     def check
-      check_this_url interpolated[:url]
+      check_this_url interpolated[:url], header_array(interpolated[:headers_to_save])
```

Comment:
@cantino I don't see how `interpolated[:headers_to_save]` could ever be `null` - even if the value in the browser is blank, it should be set to the null string, right?

Nevertheless, I've committed this change.